tests/kernel/interrupt/src/interrupt.h

```diff
@@ -0,0 +1,40 @@
+#include <ztest.h>
+
+#define IRQ_LINE(offset) (CONFIG_NUM_IRQS - ((offset) + 1))
+
+
+#if defined(CONFIG_ARM)
+#include <arch/arm/cortex_m/cmsis.h>
+static void trigger_irq(int irq)
+{
+	printk("Triggering irq : %d\n", irq);
+#if defined(CONFIG_SOC_TI_LM3S6965_QEMU) || defined(CONFIG_CPU_CORTEX_M0) \
+	|| defined(CONFIG_CPU_CORTEX_M0PLUS)
+	NVIC_SetPendingIRQ(irq);
+#else
+	NVIC->STIR = irq;
+#endif
+}
+
+#elif defined(CONFIG_RISCV32) && !defined(CONFIG_SOC_RISCV32_PULPINO)
+static void trigger_irq(int irq)
+{
+	u32_t mip;
+
+	printk("Triggering irq : %d\n", irq);
+	__asm__ volatile ("csrrs %0, mip, %1\n"
+			: "=r" (mip)
+			: "r" (1 << irq));
+}
+
+#elif defined(CONFIG_CPU_ARCV2)
+static void trigger_irq(int irq)
+{
+	printk("Triggering irq : %d\n", irq);
+	_arc_v2_aux_reg_write(_ARC_V2_AUX_IRQ_HINT, irq);
+}
+#else
+/* for not supported architecture */
+#define NO_TRIGGER_FROM_SW
```

Comment:
get rid of this.
just blacklist those targets this doesn't work on in testcase.yaml
I'm disappointed on how few targets this actually works on.
Why was x86 skipped?

Reply:
Sorry, but I am not very much clear about this comment. Trying to reply with what understood:

All achitectures have their own way, to maually trigger interrupt through software.
We have written test cases for the architectures which supports maual trigger interrupt through software. 
The way we handle or trigger irq in x86 is quite different.
Will add new interrupt test cases for x86 architecture in tests/kernel/static_idt, to avoid redundant duplicate codes.

Reply:
Again, why are you not using irq_offload()??? The whole point on why irq_offload() exists is so that we can run stuff in interrupt context in test cases.

Reply:
@andrewboie To test the nested interrupt feature we need different priority interrupts.
In this test case, first low priority ISR is getting executed, and inside I have introduced a busy loop.
Meanwhile a timer interrupt get fired and in the timer handler function we are triggering a higher priority interrupt.
As per nested interrupt rule, ISR should preempted in mid-execution if a higher priority interrupt is signaled. 
The lower priority ISR resumes execution once the higher priority ISR has completed its processing.

Now with the help of "irq_offload()" we can not trigger interrupt with different priority to full fill the test case requirement.